middleware.go

```diff
@@ -0,0 +1,28 @@
+package mux
+
+import "net/http"
+
+// MiddlewareFunc is a function which receives an http.Handler and returns another http.Handler.
+// Typically, the returned handler is a closure which does something with the http.ResponseWriter and http.Request passed
+// to it, and then calls the handler passed as parameter to the MiddlewareFunc.
+type MiddlewareFunc func(http.Handler) http.Handler
+
+// middleware interface is anything which implements a MiddlewareFunc named Middleware.
+type middleware interface {
+	Middleware(handler http.Handler) http.Handler
+}
+
+// MiddlewareFunc also implements the Middleware interface.
```

Comment:
Update this? As we don't export the `Middleware` interface (`middleware`) this may be confusing.

Reply:
Do you mean the comment or the implementation?

The implementation needs to be there in order to store `MiddlewareFunc` as `Middleware` in the `middlewares` slice. But the comment needs to be updated indeed, I slipped there :P

Reply:
Just the comment 👌🏼

Reply:
Done!